rsconf/component/devbox.py

```diff
@@ -108,34 +119,47 @@ def _gen_secrets(self, jc):
         self.secrets = PKDict({k: s[k] for k in ("host_key_f", "identity_pub_f")})
 
     def _jupyter_bashrc(self, z, path):
-        def _env(name, value):
-            v = str(value)
-            # TODO probably could recursively quote
-            if "'" in v:
-                raise ValueError(
-                    f"single quote in value={v} of bash variable name={name}"
-                )
-            self.rsconf_append(
-                path,
-                f"export {name}='{v}'",
-            )
-
         self.install_access(mode="600")
         self.install_ensure_file_exists(path)
         for n in ("package_path", "sim_types"):
             if n in z:
-                _env(f"SIREPO_FEATURE_CONFIG_{n.upper()}", ":".join(z[n]))
+                self._env(f"SIREPO_FEATURE_CONFIG_{n.upper()}", ":".join(z[n]), path)
         z.service_port = z.ssh_port + z.ssh_service_port_difference
         z.job_supervisor_port = z.service_port + z.ssh_service_port_difference
         for n in ("service_port", "job_supervisor_port"):
-            _env(f"SIREPO_PKCLI_{n.upper()}", z[n])
+            self._env(f"SIREPO_PKCLI_{n.upper()}", z[n], path)
         for n in ("DRIVER_LOCAL", "API"):
-            _env(
+            self._env(
                 f"SIREPO_JOB_{n}_SUPERVISOR_URI",
                 f"http://127.0.0.1:{z.job_supervisor_port}",
+                path,
             )
+        self._rsiviz(z, path)
 
     def _network(self, jc, z):
         n = self.buildt.get_component("network")
         z.ip = n.unchecked_public_ip() or n.ip_and_net_for_host(jc.rsconf_db.host)[0]
         n.add_public_tcp_ports([str(z.ssh_port)])
+
+    def _rsiviz(self, z, path):
+        u = z.users[self.user_name]
+        if not isinstance(u, PKDict) or not "rsiviz" in u:
+            return
+        self._env(
+            "PYKERN_PKASYNCIO_SERVER_PORT",
+            u.rsiviz.port_base,
+            path,
+        )
+        self._env(
+            "RSIVIZ_PKCLI_SERVICE_DICE_NETWORK_CLUSTER_INTERFACE_ADDRESS",
+            ipaddress.ip_address(u.rsiviz.ip_base) + 1,
```

Comment:
This is really confusing, but I guess it works...

Reply:
Any suggestions on how to make it less confusing? We just need to be on different networks and this seemed like a simply way to accomplish that.

Reply:
I meant the fact that `ip_address() + 1` does something (not sure what). I don't like overloading in this way. It is going to add 1 to the ip address. Not that confusing I guess.